File: backend/src/main/java/com/malahot/entity/Competition.java
package com.malahot.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("competition")
public class Competition {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String gameType;

    private String description;

    private String rules;

    private String coverImage;

    private Integer maxTeams;

    private Integer teamSize;

    private String format;

    private LocalDateTime registrationStart;

    private LocalDateTime registrationEnd;

    private LocalDateTime competitionStart;

    private String status;

    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/competition.ts
import api from './index';

export interface Competition {
  id: number;
  title: string;
  gameType: string;
  description: string;
  rules: string;
  coverImage: string | null;
  maxTeams: number;
  teamSize: number;
  format: string;
  registrationStart: string;
  registrationEnd: string;
  competitionStart: string | null;
  status: string;
  createdBy: number;
  createdAt: string;
}

export interface PageResult<T> {
  records: T[];
  total: number;
  size: number;
  current: number;
  pages: number;
}

export interface CompetitionRequest {
  title: string;
  gameType: string;
  description?: string;
  rules?: string;
  coverImage?: string;
  maxTeams: number;
  teamSize: number;
  format?: string;
  registrationStart: string;
  registrationEnd: string;
  competitionStart?: string;
}

export const competitionApi = {
  list: (params: { page?: number; size?: number; status?: string; gameType?: string }) =>
    api.get<any, { data: PageResult<Competition> }>('/competitions', { params }),
  getById: (id: number) =>
    api.get<any, { data: Competition }>(`/competitions/${id}`),
  create: (data: CompetitionRequest) =>
    api.post<any, { data: Competition }>('/competitions', data),
  update: (id: number, data: CompetitionRequest) =>
    api.put<any, { data: Competition }>(`/competitions/${id}`, data),
  updateStatus: (id: number, status: string) =>
    api.put(`/competitions/${id}/status`, null, { params: { status } }),
};
